Modix.Services/ErrorHandling/InsufficientRankResult.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Discord;
+using Modix.Common.ErrorHandling;
+using Modix.Data.Models.Core;
+
+namespace Modix.Services.ErrorHandling
+{
+    /// <summary>
+    /// A <see cref="ServiceResult"/> representing a failure to execute a request due to insufficient rank
+    /// </summary>
+    public class InsufficientRankResult : ServiceResult
+    {
+        /// <summary>
+        /// The Id of the required role
+        /// </summary>
+        public ulong RequiredRoleId { get; private set; }
```

Comment:
Remove the `private set`, please, if you're only setting these in constructors. Makes it clearer that these are actually immutable (and gets the compiler to enforce it).

Reply:
Fixed